bundles-lib/src/main/java/org/apache/metron/bundles/bundle/BundleDetails.java

```diff
@@ -0,0 +1,191 @@
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Metadata about a bundle.
+ * the withCoordinates and withBundleFile properties are required
+ */
+public class BundleDetails {
+
+    private final FileObject bundleFile;
+
+    private final BundleCoordinates coordinates;
+    private final BundleCoordinates dependencyCoordinates;
+
+    private final String buildTag;
+    private final String buildRevision;
+    private final String buildBranch;
+    private final String buildTimestamp;
+    private final String buildJdk;
+    private final String builtBy;
+
+    private BundleDetails(final Builder builder) {
+        this.bundleFile = builder.bundleFile;
+        this.coordinates = builder.coordinates;
+        this.dependencyCoordinates = builder.dependencyCoordinates;
+
+        this.buildTag = builder.buildTag;
+        this.buildRevision = builder.buildRevision;
+        this.buildBranch = builder.buildBranch;
+        this.buildTimestamp = builder.buildTimestamp;
+        this.buildJdk = builder.buildJdk;
+        this.builtBy = builder.builtBy;
+
+        if (this.coordinates == null) {
+            if (this.bundleFile == null) {
+                throw new IllegalStateException("Coordinate cannot be null");
+            } else {
+                throw new IllegalStateException("Coordinate cannot be null for " + this.bundleFile.getName());
+            }
+        }
+
+        if (this.bundleFile == null) {
+            throw new IllegalStateException("bundleFile cannot be null for " + this.coordinates
+                .getId());
+        }
+    }
+
+    public FileObject getBundleFile() {
+        return bundleFile;
+    }
+
+    public BundleCoordinates getCoordinates() {
+        return coordinates;
+    }
+
+    public BundleCoordinates getDependencyCoordinates() {
+        return dependencyCoordinates;
+    }
+
+    public String getBuildTag() {
+        return buildTag;
+    }
+
+    public String getBuildRevision() {
+        return buildRevision;
+    }
+
+    public String getBuildBranch() {
+        return buildBranch;
+    }
+
+    public String getBuildTimestamp() {
+        return buildTimestamp;
+    }
+
+    public String getBuildJdk() {
+        return buildJdk;
+    }
+
+    public String getBuiltBy() {
+        return builtBy;
+    }
+
+    @Override
+    public String toString() {
+        return coordinates.toString();
+    }
+
+    public Date getBuildTimestampDate() {
+        if (buildTimestamp != null && !buildTimestamp.isEmpty()) {
+            try {
+                SimpleDateFormat buildTimestampFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+                Date buildTimestampDate = buildTimestampFormat.parse(buildTimestamp);
+                return buildTimestampDate;
+            } catch (ParseException parseEx) {
+                return null;
+            }
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Builder for BundleDetails.
```

Comment:
Please also javadoc the required properties on the Builder, since that the public i/f that actually gets used.
> + * the withCoordinates and withBundleFile properties are required